PWGHF/D2H/Tasks/taskD0.cxx

```diff
@@ -48,11 +50,23 @@ struct HfTaskD0 {
   using D0CandidatesKF = soa::Join<D0Candidates, aod::HfCand2ProngKF>;
   using D0CandidatesMcKF = soa::Join<D0CandidatesKF, aod::HfCand2ProngMcRec>;
 
+  using D0CandidatesMl = soa::Join<D0Candidates, aod::HfMlD0>;
+  using D0CandidatesMlMc = soa::Join<D0CandidatesMl, aod::HfCand2ProngMcRec>;
+  using D0CandidatesMlKF = soa::Join<D0CandidatesMl, aod::HfCand2ProngKF>;
+  using D0CandidatesMlMcKF = soa::Join<D0CandidatesMlKF, aod::HfCand2ProngMcRec>;
+
   Partition<D0Candidates> selectedD0Candidates = aod::hf_sel_candidate_d0::isSelD0 >= selectionFlagD0 || aod::hf_sel_candidate_d0::isSelD0bar >= selectionFlagD0bar;
   Partition<D0CandidatesKF> selectedD0CandidatesKF = aod::hf_sel_candidate_d0::isSelD0 >= selectionFlagD0 || aod::hf_sel_candidate_d0::isSelD0bar >= selectionFlagD0bar;
   Partition<D0CandidatesMc> selectedD0CandidatesMc = aod::hf_sel_candidate_d0::isRecoHfFlag >= selectionFlagHf;
   Partition<D0CandidatesMcKF> selectedD0CandidatesMcKF = aod::hf_sel_candidate_d0::isRecoHfFlag >= selectionFlagHf;
 
+  Partition<D0CandidatesMl> selectedD0CandidatesMl = aod::hf_sel_candidate_d0::isSelD0 >= selectionFlagD0 || aod::hf_sel_candidate_d0::isSelD0bar >= selectionFlagD0bar;
+  Partition<D0CandidatesMlKF> selectedD0CandidatesMlKF = aod::hf_sel_candidate_d0::isSelD0 >= selectionFlagD0 || aod::hf_sel_candidate_d0::isSelD0bar >= selectionFlagD0bar;
+  Partition<D0CandidatesMlMc> selectedD0CandidatesMlMc = aod::hf_sel_candidate_d0::isRecoHfFlag >= selectionFlagHf;
+  Partition<D0CandidatesMlMcKF> selectedD0CandidatesMlMcKF = aod::hf_sel_candidate_d0::isRecoHfFlag >= selectionFlagHf;
+
+  HistogramConfigSpec hTHnBdtScoreVsMassVsPtVsYVsOrigin{HistType::kTHnSparseD, {{100, 0.f, 1.f}, {100, 0.f, 1.f}, {500, 0., 5.}, {360, 0., 36.}, {100, -5., 5.}, {3, -0.5, 2.5}}};
```

Comment:
I have 2 suggestions here:
 1. the axis of the mass can be already setup with a reasonable range and binning, instead of `{500, 0., 5.}`. You shoul center it to the D0 mass (e.g. you can get inspiration from the histograms defined below, for example `{120, 1.5848, 2.1848}`)
 2. in this way in MC you do not have any possibility to understand if your signal candidate was D0 or D0bar. What about adding another axis storying this info?

Reply:
hi @atavirag you still miss the info in MC to distinguish the reflections, i.e. the MC truth that tells you if you have a real D0 or D0bar, am I right?